src/sagas/wallet.js

```diff
@@ -568,16 +567,19 @@ export function* onWalletConnStateUpdate({ payload }) {
 }
 
 export function* onWalletReloadData() {
-  yield put(onStartWalletLock());
-
+  const useWalletService = yield select((state) => state.useWalletService);
   const wallet = yield select((state) => state.wallet);
 
-  // Since we close the channel after a walletReady event is received,
-  // we must fork this saga again so we setup listeners again.
-  yield fork(listenForWalletReady, wallet);
+  // If we are using the wallet-service, we don't need to wait until the addresses
+  // are reloaded since they are stored on the wallet-service itself.
+  if (!useWalletService) {
+    // Since we close the channel after a walletReady event is received,
+    // we must fork this saga again so we setup listeners again.
+    yield fork(listenForWalletReady, wallet);
```

Comment:
I understand we don't need to wait for the wallet ready to continue if it's wallet service but don't we need this listener anyway for other things, e.g., error handling when the load fails, ws connection lost (this last one I feel it's not affected)?